Docs/source/usage/python.rst

```diff
@@ -145,6 +145,56 @@ The input file should have the line ``sim.step()`` which runs the simulation.
 where ``<n_ranks>`` is the number of MPI ranks used, and ``<python_script>``
 is the name of the script.
 
+When running WarpX directly from Python it is possible to interact with the simulation
+by installing ``CallbackFunctions``, which will execute a given Python function at a
+specific location in the WarpX simulation loop.
+
+.. autoclass:: pywarpx.callbacks.CallbackFunctions
+
+Places in the WarpX loop where callbacks are available include:
+``afterinit``, ``beforecollisions``, ``aftercollisions``, ``beforeEsolve``, ``afterEsolve``,
+``beforedeposition``, ``afterdeposition``, ``beforestep``, ``afterstep``, ``afterdiagnostics``,
+``afterrestart`` and ``oncheckpointsignal``.
+
+There are several "hooks" available via the ``libwarpx`` shared library to access and manipulate
+simulation objects (particles, fields and memory buffers) as well as general properties
+(such as processor number). These "hooks" are accessible through the `Simulation.extension` object.
+
+.. autofunction:: pywarpx.libwarpx.getNProcs
+
+.. autofunction:: pywarpx.picmi.Simulation.extension.getNProcs
```

Comment:
@ax3l and @dpgrote please let me know which of the above forms you guys prefer. The intent is that users always interact with `libwarpx` through the `simulation` object, right? For that reason I think referencing the functions as part of `Simulation.extension` makes sense.

Reply:
I agree that the latter is better since the is the recommended way of accessing the functions.